src/couch_httpd_auth.erl

```diff
@@ -372,9 +372,11 @@ maybe_value(Key, Else, Fun) ->
 
 maybe_upgrade_password_hash(Req, UserName, Password, UserProps,
         AuthModule, AuthCtx) ->
+    Upgrade = config:get("couch_httpd_auth", "upgrade_password_on_auth", "true"),
```

Comment:
[config:get_boolean/3](https://github.com/apache/couchdb-config/blob/master/src/config.erl#L102-L108) is better here.